primefaces/src/main/java/org/primefaces/component/inputnumber/InputNumberRenderer.java

```diff
@@ -391,7 +394,24 @@ private String getMaximum(InputNumber inputNumber, Object value) {
     }
 
     private boolean isIntegral(Object value) {
-        return value instanceof Long || value instanceof Integer || value instanceof Short || value instanceof BigInteger || value instanceof Byte;
+        return value instanceof Long
+                || value instanceof Integer
+                || value instanceof Short
+                || value instanceof BigInteger
+                || value instanceof Byte;
+    }
+
+    private boolean isIntegral(FacesContext context, InputNumber inputNumber) {
+        ValueExpression valueExpression = inputNumber.getValueExpression("value");
+        if (valueExpression == null) {
+            return false;
+        }
+        Class<?> type = valueExpression.getType(context.getELContext());
+        return type.isAssignableFrom(Long.class)
```

Comment:
Maybe you can simplify by moving these conditions in isIntegral(Object value) and then you can directly call in isIntegral(FacesContext context, InputNumber inputNumber)?

Reply:
Agreed!   Other than that looks great

Reply:
Right, it's better now. I also optimized it a bit (it was called three times in a row)